rpc-server/src/utils.rs

```diff
@@ -171,11 +171,66 @@ where
+        // separate mismatching successful and failure responses into different targets
+        // to make it easier to find reasons of the mismatching
+        // let reason = String::new();
+        // let description = String::new();
+        let (reason, description) = if read_rpc_response_is_ok && near_rpc_response_is_ok {
+            // Both services(read_rpc and near_rpc) have a successful response
+            // if data mismatch we are logging into `shadow_data_consistency_successful_responses`
+            // both response objects for future investigation.
+            (
+                "shadow_data_consistency_successful_responses",
+                format!(
+                    "Shadow data check: DATA MISMATCH\n READ_RPC_DATA: {:?}\n NEAR_RPC_DATA: {:?}\n",
+                    read_rpc_json,
+                    near_rpc_json,
+                )
+            )
+        } else if !read_rpc_response_is_ok && near_rpc_response_is_ok {
+            // read_rpc service has failure response
+            // and near_rpc has successful response we are logging into
+            // `shadow_data_consistency_read_rpc_failure_response` just message about mismatch
+            // data without objects.
+            // In the future we should be investigate why we got error in the read_rpc.
+            (
+                "shadow_data_consistency_read_rpc_failure_response",
+                String::from(
+                    "Shadow data check: read_rpc response failed, near_rpc response success",
+                ),
+            )
+        } else if read_rpc_response_is_ok && !near_rpc_response_is_ok {
+            // read_rpc service has successful response
+            // and near_rpc has failure response we are logging into
+            // `shadow_data_consistency_near_failure_response` just message about mismatch
+            // data without objects. Expected that all error will be related with network issues.
+            (
+                "shadow_data_consistency_near_failure_response",
+                String::from(
+                    "Shadow data check: read_rpc response success, near_rpc response failed",
+                ),
+            )
+        } else {
+            // Both services(read_rpc and near_rpc) have a failure response
+            // we are logging into `shadow_data_consistency_failure_responses`
+            // both response objects for future investigation.
+            // expected we will only have a difference in the error text.
+            (
+                "shadow_data_consistency_failure_responses",
+                format!(
+                    "Shadow data check: DATA MISMATCH\n READ_RPC_DATA: {:?}\n NEAR_RPC_DATA: {:?}\n",
+                    read_rpc_json,
+                    near_rpc_json,
+                ),
+            )
+        };
+        return Err(ShadowDataConsistencyError::ResultsDontMatch(format!(
+            "{err}. Reason: {reason}. Description {description}"
```

Comment:
Can you post an example of such input for clarity, please?

Reply:
[exampl.log](https://github.com/near/read-rpc/files/12237671/exampl.log)